File: backend/app/schemas/gamification.py
from datetime import datetime

from pydantic import BaseModel


class AchievementCatalogEntry(BaseModel):
    code: str
    title: str
    description: str
    icon: str
    unlocked: bool
    unlocked_at: datetime | None


class GamificationProfileRead(BaseModel):
    xp_total: int
    level: int
    xp_for_next_level: int
    level_progress_percent: float
    current_streak: int
    longest_streak: int
    achievements: list[AchievementCatalogEntry]

File: frontend/src/types/gamification.ts
export interface AchievementCatalogEntry {
  code: string
  title: string
  description: string
  icon: string
  unlocked: boolean
  unlocked_at: string | null
}

export interface GamificationProfile {
  xp_total: number
  level: number
  xp_for_next_level: number
  level_progress_percent: number
  current_streak: number
  longest_streak: number
  achievements: AchievementCatalogEntry[]
}
